src/utils/StringUtils.ts

```diff
@@ -122,6 +124,32 @@ export class StringUtils {
     return str.charAt(0).toUpperCase() + str.slice(1);
   }
 
+  static buildStringTemplateFromResult(template: string, result: IQueryResult) {
+    if (!template) {
+      return '';
+    }
+    return template.replace(/\$\{(.*?)\}/g, (value: string) => {
+      let key = value.substring(2, value.length - 1);
+      let newValue = StringUtils.readFromObject(result, key);
+      if (!newValue) {
+        newValue = StringUtils.readFromObject(window, key);
+      }
+      if (!newValue) {
+        new Logger(this).warn(`${key} used in the ResultLink template is undefined for this result: ${result.title}`);
+      }
+      return newValue || value;
+    });
+  }
+
+  private static readFromObject(object: Object, key: string): string {
+    if (object && key.indexOf('.') !== -1) {
```

Comment:
Since `key.indexOf('.')` is reused, it would be nice to store the value in a variable. e.g.

```
const firstPeriodIndex = key.indexOf('.')
```

Reply:
I copy/pasted that function, I can do that, no worries.